src/Util/ZipFiles.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Util;
+
+use App\Entity\File;
+
+use Doctrine\Common\Collections\Collection;
+
+use ZipArchive;
+
+class ZipFiles
+{
+
+    /**
+     * ZipArchive class instance.
+     *
+     * @var ZipArchive
+     */
+    private $zip;
+
+    /**
+     * ZipFiles constructor.
+     */
+    public function __construct()
+    {
+        $this->zip = new ZipArchive();
+    }
+
+    /**
+     * Creates a zip file from collection of files.
+     *
+     * @param Collection|File $files   Files that need to be zipped.
+     * @param string          $zipFile Filename of the zipfile.
+     *
+     * @throws \Exception When utility class can not open/write to zip file.
+     *
+     * @return string
+     */
+    public function createZipFile(Collection $files, string $zipFile) : string
+    {
+        $flag = (file_exists($zipFile))? ZIPARCHIVE::OVERWRITE : ZIPARCHIVE::CREATE;
+        if ($this->zip->open($zipFile, $flag) === true) {
```

Comment:
I'm concerned there really isn't a way to control where the zipfile itself is created. I'm assuming it would go into the symfony var folder, but it would be nice if we could contain them to a separate filesystem that could be filled without detriment to the rest of the system. If var itself fills, I don't think the site would be unusable.

Reply:
I think we should do that in the messageHandler class which calls this utility